controllers/imageclusterinstall_controller.go

```diff
@@ -577,30 +608,22 @@ func (r *ImageClusterInstallReconciler) getBMH(ctx context.Context, bmhRef *v1al
 	return bmh, nil
 }
 
-func (r *ImageClusterInstallReconciler) removeBMHImage(ctx context.Context, bmhRef *v1alpha1.BareMetalHostReference) (bool, error) {
-	bmh := &bmh_v1alpha1.BareMetalHost{}
+func (r *ImageClusterInstallReconciler) removeBMHDataImage(ctx context.Context, bmhRef *v1alpha1.BareMetalHostReference) (bool, error) {
+	dataImage := &bmh_v1alpha1.DataImage{}
+
 	key := types.NamespacedName{
 		Name:      bmhRef.Name,
 		Namespace: bmhRef.Namespace,
 	}
-	if err := r.Get(ctx, key, bmh); err != nil {
-		return false, err
-	}
-	patch := client.MergeFrom(bmh.DeepCopy())
 
-	dirty := false
-	if bmh.Spec.Image != nil {
-		bmh.Spec.Image = nil
-		dirty = true
-	}
-
-	if dirty {
-		if err := r.Patch(ctx, bmh, patch); err != nil {
-			return false, err
+	if err := r.Get(ctx, key, dataImage); err != nil {
+		if apierrors.IsNotFound(err) {
+			return true, nil
 		}
+		return false, err
 	}
 
-	return dirty, nil
+	return true, r.Client.Delete(ctx, dataImage)
```

Comment:
Same here, what happens to the BMH when we remove the data image?
* I'm guessing nothing changes